src/compiler/utilities.ts

```diff
@@ -2547,6 +2499,52 @@ namespace ts {
         }
     }
 
+    /* @internal */
+    export function getBinaryOperatorPrecedence(kind: SyntaxKind): number {
+        switch (kind) {

```

Comment:
minor nit: It's a bit odd to read `getOperatorPrecedence` in descending order then see `getBinaryOperatorPrecedence` in ascending order. We might want to change the order of one or the other to be consistent.